src/routes/InstancesPage/InstancesToolbarSearchFilter.js

```diff
@@ -87,6 +87,20 @@ function InstancesToolbarSearchFilter({ filters, setFilters }) {
     onSelect('status', event, selection);
   }
 
+  function applyCurrentSearchText(field, input, setInput) {
+    if (!input) return;
+
+    setFilters((prevFilters) => {
+      const newFilters = { ...prevFilters };
```

Comment:
iirc the spread operator creates a shallow copy -- is `prevFilters` guaranteed to be shallow?

Reply:
Oh good call - I'll swap this to a `structuredClone`.